lib/Optimizer/LatePasses/CopyLoweringPass.cpp

```diff
@@ -37,6 +40,24 @@ class CopyLoweringPass : public gil::InstVisitor<CopyLoweringPass> {
             return;
         }
 
+        // Check for infinite recursion: if we're inside the copy function
+        // for this struct, warn about potential infinite recursion
+        ast::FunctionDecl *copyFunc = structure->getDecl()->getCopyFunction();
+        if (_currentFunction && _currentFunction->getDecl() == copyFunc) {
```

Comment:
Can't you use loadInst->getParent()->getParent() instead of storing the _currentFunction as a field?

Reply:
?????????? no